### PR TITLE
Test mode

### PR DESCRIPTION
**Pull request by vandevoorde**
Add config options testMode and localServerPort

### CODE REVIEW

File: analytics-samples/analytics-sample/src/main/java/com/bytegain/analytics/sample/SampleApp.java [hunk 1]
```diff
@@ -28,6 +28,8 @@ public void onCreate() {
     // Initialize a new instance of the Analytics client.
     Analytics.Builder builder =
         new Analytics.Builder(this, ANALYTICS_API_KEY)
+//            .testMode()
+//            .localServerPort(5001)
```
Comment (quirkyllama) [inline analytics-samples/analytics-sample/src/main/java/com/bytegain/analytics/sample/SampleApp.java]: Do we want to expose this? Customers will never have a localserver

File: analytics/src/main/java/com/bytegain/analytics/ConnectionFactory.java [hunk 2]
```diff
@@ -36,8 +45,9 @@ public HttpURLConnection projectSettings(String writeKey) throws IOException {
    */
   public HttpURLConnection upload(String writeKey) throws IOException {
     // 10.0.2.2 is emulator's route to host's 127.0.0.1
+    final String localUrl = String.format(Locale.US, "http://10.0.2.2:%d/v1/batch", localServerPort);
```
Comment (quirkyllama) [inline analytics/src/main/java/com/bytegain/analytics/ConnectionFactory.java:48]: WHy no tmake the who URL a config param?

**Review by quirkyllama — APPROVED** [review]

File: analytics-samples/analytics-sample/src/main/java/com/bytegain/analytics/sample/SampleApp.java (hunk 1 above)
Comment (vandevoorde) [reply analytics-samples/analytics-sample/src/main/java/com/bytegain/analytics/sample/SampleApp.java]: Yeah, but I do this so infrequently that I'll forget.   I'll add a comment that it's for ByteGain only.

File: analytics/src/main/java/com/bytegain/analytics/ConnectionFactory.java (hunk 2 above)
Comment (vandevoorde) [reply analytics/src/main/java/com/bytegain/analytics/ConnectionFactory.java:48]: I haven't needed it yet.  We could make it the full URL if we want to be able to set up a test server (staging?) for customers.

Should I change the iOS version too?  It's just a port.

Comment (quirkyllama) [reply analytics/src/main/java/com/bytegain/analytics/ConnectionFactory.java:48]: Fine for now, just wondering if there was something specific driving this

**Pull request merged**